### ISSUE TITLE
Issue: Inconsistency in summary table: "no-op" vs `RuntimeError`

### ISSUE THREAD

**Issue by nghtctrl** [pytorch/tutorials#3682]
On [Understanding requires_grad, retain_grad, Leaf, and Non-leaf Tensors](https://docs.pytorch.org/tutorials/beginner/understanding_leaf_vs_nonleaf_tutorial.html), I noticed an inconsistency between the tutorial text and the summary table regarding the behavior of `retain_grad()` on tensors where `requires_grad=False`.

The tutorial text correctly states:

> If we call `retain_grad()` on a node that has `requires_grad=False`, PyTorch actually throws an error, since it can’t store the gradient if it is never calculated.

However, the first row of the summary table shows:

<img width="877" height="164" alt="Image" src="https://github.com/user-attachments/assets/3c78ba95-3077-4f5b-abce-a42c85f15bb8" />

This is inaccurate because an error is not a "no-op"; they are conceptually different. Therefore, the summary table should replace "no-op" with `RuntimeError` to accurately reflect what the tutorial text says.

**Comment by jeehoh** [comment]
Closing this because it has already been raised on #3652
